source/Octopus.Tentacle.Kubernetes.Tests.Integration/Setup/KubernetesAgentInstaller.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System.Diagnostics;
+using System.Reflection;
+using System.Text;
+using Octopus.Client.Model;
+using Octopus.Tentacle.CommonTestUtils;
+using Octopus.Tentacle.CommonTestUtils.Logging;
+using Octopus.Tentacle.Security.Certificates;
+using Octopus.Tentacle.Util;
+
+namespace Octopus.Tentacle.Kubernetes.Tests.Integration.Setup;
+
+public class KubernetesAgentInstaller
+{
+    readonly string helmExePath;
+    readonly string kubeCtlExePath;
+    readonly TemporaryDirectory temporaryDirectory;
+    readonly ILogger logger;
+    readonly string kubeConfigPath;
+
+    bool isAgentInstalled;
+
+    public KubernetesAgentInstaller(TemporaryDirectory temporaryDirectory, string helmExePath, string kubeCtlExePath, string kubeConfigPath, ILogger logger)
+    {
+        this.temporaryDirectory = temporaryDirectory;
+        this.helmExePath = helmExePath;
+        this.kubeCtlExePath = kubeCtlExePath;
+        this.kubeConfigPath = kubeConfigPath;
+        this.logger = logger;
+
+        AgentName = Guid.NewGuid().ToString("N");
+    }
+
+    public string AgentName { get; }
+
+    string Namespace => $"octopus-agent-{AgentName}";
+
+    public Uri SubscriptionId { get; } = PollingSubscriptionId.Generate();
+
+    public async Task<string> InstallAgent(int listeningPort)
+    {
+        var valuesFilePath = await WriteValuesFile(listeningPort);
+        var arguments = BuildAgentInstallArguments(valuesFilePath);
+
+        var sw = new Stopwatch();
+        sw.Restart();
+
+        var sb = new StringBuilder();
+        var sprLogger = new LoggerConfiguration()
+            .WriteTo.Logger(logger)
+            .WriteTo.StringBuilder(sb)
+            .MinimumLevel.Debug()
+            .CreateLogger();
+
+        var exitCode = SilentProcessRunner.ExecuteCommand(
+            helmExePath,
+            arguments,
+            temporaryDirectory.DirectoryPath,
+            sprLogger.Debug,
+            sprLogger.Information,
+            sprLogger.Error,
+            CancellationToken.None);
+
+        sw.Stop();
+
+        if (exitCode != 0)
+        {
+            throw new InvalidOperationException($"Failed to install Kubernetes Agent via Helm. Logs: {sb}");
+        }
+
+        isAgentInstalled = true;
+
+        var thumbprint = await GetAgentThumbprint();
+
+        logger.Information("Agent certificate thumbprint: {Thumbprint:l}", thumbprint);
+
+        return thumbprint;
+    }
+
+    async Task<string> WriteValuesFile(int listeningPort)
+    {
+        var asm = Assembly.GetExecutingAssembly();
+        var valuesFileName = asm.GetManifestResourceNames().First(n => n.Contains("agent-values.yaml", StringComparison.OrdinalIgnoreCase));
+        using var reader = new StreamReader(asm.GetManifestResourceStream(valuesFileName)!);
+
+        var valuesFile = await reader.ReadToEndAsync();
+
+        var serverCommsAddress = $"https://dockerhost.default.svc.cluster.local:{listeningPort}";
```

Comment:
Can `dockerhost` be pulled to a const

Reply:
it can, we don't use it anywhere else though

Reply:
I think it'll help communicate that `dockerhost` is a special name we made up

Reply:
sure!

Reply:
done in [a54f7ec](https://github.com/OctopusDeploy/OctopusTentacle/pull/888/commits/a54f7ecb32e1db81c0d037bbb03101fa118d919f) and commented it